packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,456 @@
+  QuanticRecommendationList,
+  'c-quantic-recommendation-list',
+  defaultOptions
+);
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildRecommendationList: functionMocks.buildRecommendationList,
+      buildResultTemplatesManager: functionMocks.buildResultTemplatesManager,
+      loadFieldActions: functionMocks.loadFieldActions,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticRecommendationList && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticRecommendationList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+describe('c-quantic-recommendation-list', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+    console.error = jest.fn();
+  });
+
+  afterEach(() => {
+    cleanup();
+    isInitialized = false;
+    recommendationListState = initialRecommendationListState;
+  });
+
+  describe('when the component is loading', () => {
+    it('should display the placeholder component when loading', async () => {
+      recommendationListState = {
+        isLoading: true,
+        recommendations: [],
+        error: null,
+        searchResponseId: null,
+      };
+      const element = createTestComponent();
+      await flushPromises();
+
+      const loadingPlaceholder = element.shadowRoot.querySelector(
+        selectors.placeholder
+      );
+
+      expect(loadingPlaceholder).not.toBeNull();
+    });
+
+    it('should not display the placeholder component when there is an error in the state', async () => {
+      recommendationListState = {
+        isLoading: false,
+        recommendations: [],
+        error: {
+          statusCode: 500,
+          message: 'Error loading recommendations',
+        },
+        searchResponseId: null,
+      };
+      const element = createTestComponent();
+      await flushPromises();
+
+      const loadingPlaceholder = element.shadowRoot.querySelector(
+        selectors.placeholder
+      );
+
+      expect(loadingPlaceholder).toBeNull();
+    });
+  });
+
+  describe('controller initialization', () => {
+    it('should build the buildRecommendationList controller with the proper parameters and subscribe to its state', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.buildRecommendationList).toHaveBeenCalledTimes(1);
+      expect(functionMocks.buildRecommendationList).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          options: {
+            id: defaultOptions.recommendation,
+            numberOfRecommendations: Number(
```

Comment:
why Number(...) ? `defaultOptions.numberOfRecommendations` is already a number so no need.

In the code of the component you may find `Number(this.numberOfRecommendations)`  because public properties of the component can be passed as strings: 
```
<c-quantic-recomendation-list number-of-recommendations ="5"></c-quantic-recomendation-list>
```
always try to challenge the reasoning behind every piece of code you write 👍